src/Microsoft.Graph.Core/Requests/Middleware/OdataQueryHandler.cs

```diff
@@ -0,0 +1,61 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph
+{
+    using Microsoft.Kiota.Http.HttpClientLibrary.Extensions;
+    using System;
+    using System.Net.Http;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// A <see cref="DelegatingHandler"/> implementation that handles compression.
+    /// </summary>
+    public class OdataQueryHandler : DelegatingHandler
+    {
+        private readonly ODataQueryHandlerOption odataQueryHandlerOption;
+        private readonly Regex odataQueryRegex = new Regex("(?i)([^$])(count|deltatoken|expand|filter|format|orderby|search|select|skip|skiptoken|top)=",RegexOptions.Compiled);
```

Comment:
nit: dotnet developers usually use regexOptions.ignoreCase instead of the prag match syntax